src/dispatch/task/flows.py

```diff
@@ -100,76 +97,41 @@ def send_task_notification(
 
 def create_or_update_task(db_session, incident, task: dict, notify: bool = False):
     """Creates a new task in the database or updates an existing one."""
-    incident_task = task_service.get_by_resource_id(db_session=db_session, resource_id=task["id"])
-
-    assignees = []
-    for a in task["assignees"]:
-        assignees.append(
-            db_session.merge(
-                incident_add_or_reactivate_participant_flow(
-                    a, incident_id=incident.id, db_session=db_session
-                )
-            )
-        )
-
-    description = task["description"][0]
-    status = TaskStatus.open if not task["status"] else TaskStatus.resolved
-    resource_id = task["id"]
-    weblink = task["web_link"]
-
-    # TODO we can build this out as our scraping gets more advanced
-    tickets = [
-        ticket_service.get_or_create_by_weblink(db_session=db_session, weblink=t["web_link"])
-        for t in task["tickets"]
-    ]
+    incident_task = task_service.get_by_resource_id(
+        db_session=db_session, resource_id=task["resource_id"]
+    )
 
     if incident_task:
-        # allways update tickets and assignees
-        incident_task.assignees = assignees
-        incident_task.tickets = tickets
-
         # only notify if it's newly resolved
-        if status == TaskStatus.resolved:
+        prev_status = incident_task.status
+        task = task_service.update(
+            db_session=db_session, task=incident_task, task_in=TaskUpdate(**task)
+        )
+        if prev_status == TaskStatus.resolved:
             if incident_task.status != TaskStatus.resolved:
```

Comment:
I'm confused about this. If I'm reading it correctly, we're setting the status of the task to resolved when we go from resolved to open status, assuming `incident_task.status` has the updated status, and then if notify is `True` we send a notification that the task has been resolved? Shouldn't we check if the new status is resolved and the previous one open, and then notify if notify is `True`?

Also, renaming `task` to `updated_incident_task` and using that instead of `incident_task` would make it easier to read.

Reply:
afaik, I don't believe the logic was changed. I added a few comments to hopefully make the logic a bit clearer.